src/main/java/org/opentripplanner/middleware/controllers/api/TripHistoryController.java

```diff
@@ -41,11 +48,58 @@
 
+    private final String ROOT_ROUTE;
+
+    public TripHistoryController(String apiPrefix) {
+        this.ROOT_ROUTE = apiPrefix + "secure/triprequests";
+    }
+
+    /**
+     * Register the API endpoint and GET resource to get trip requests
+     * when spark-swagger calls this function with the target API instance.
+     */
+    @Override
+    public void bind(final SparkSwagger restApi) {
+        ApiEndpoint apiEndpoint = restApi.endpoint(
+            endpointPath(ROOT_ROUTE).withDescription("Interface for retrieving trip requests."),
+            (q, a) -> LOG.info("Received request for 'triprequests' Rest API")
+        );
+        apiEndpoint
+            .get(path(ROOT_ROUTE)
+                    .withDescription("Gets a list of all trip requests for a user.")
+                    .withQueryParam()
+                        .withName("userId")
+                        .withRequired(true)
+                        .withDescription("The OTP user for which to retrieve trip requests.").and()
+                    .withQueryParam()
+                        .withName(LIMIT_PARAM_NAME)
+                        .withDefaultValue(String.valueOf(DEFAULT_LIMIT))
+                        .withDescription("If specified, the maximum number of trip requests to return, starting from the most recent.").and()
+                    .withQueryParam()
+                        .withName(FROM_DATE_PARAM_NAME)
+                        .withPattern("yyyy-MM-dd")
+                        .withDefaultValue("The current date")
+                        .withDescription("If specified, the earliest date (format yyyy-MM-dd) for which trip requests are retrieved.").and()
+                    .withQueryParam()
+                        .withName(TO_DATE_PARAM_NAME)
+                        .withPattern("yyyy-MM-dd")
+                        .withDefaultValue("The current date")
+                        .withDescription("If specified, the latest date (format yyyy-MM-dd) for which usage logs are retrieved.").and()
+                    .withProduces(MIMETYPES_JSONONLY)
+                    // Note: unlike the name suggests, withResponseAsCollection does not generate an array
+                    // as the return type for this method. (It does generate the type for that class nonetheless.)
+                    .withResponseAsCollection(TripRequest.class),
+                TripHistoryController::getTripRequests, JsonUtils::toJson)
+
+            // Options response for CORS
+            .options(path(""), (req, res) -> "");
+    }
+
     /**
-     * Return JSON representation of a user's trip request history based on provided parameters.
+     * Return a user's trip request history based on provided parameters.
      * An authorized user (Auth0) and user id are required.
      */
-    public static String getTripRequests(Request request, Response response) {
+    public static List<TripRequest> getTripRequests(Request request, Response response) {
```

Comment:
Access can be private.

Reply:
Fixed in https://github.com/ibi-group/otp-middleware/pull/55/commits/281ef43adac1a5e3141e9088ad41d0b4310af57f